Added 5 and changed version for 1

ambrishrawat opened the pull request:
Added the following 5
- deepwalk==1.0.3
- imblearn==0.0
- gensim==3.8.3
- tensorboard-logger==0.1.0
- futures==2.1.6

Our code isn't working with igraph 0.10.2, but works with 0.9.11. 
@jayqi : I believe this was added my another team so am not sure if the version change breaks stuff for them. Please suggest what would be the best way forward

szhorvat commented:
If you found problems in python-igraph, please report them, https://github.com/igraph/python-igraph/issues  The great majority of behaviour changes between 0.9 and 0.10 are the result of bugfixes, the rest usually have easy workarounds.

jayqi commented on runtime/environment-cpu.yml:
```diff
@@ -38,3 +38,8 @@ dependencies:
       - tensorflow_privacy==0.7.3
       - tensorflow-probability==0.15.0
+      - deepwalk==1.0.3
+      - imblearn==0.0
```
[imblearn](https://pypi.org/project/imblearn/) looks like it's just a dummy package that points to imbalanced-learn. That has already been added to the environment. 

https://github.com/drivendataorg/pets-prize-challenge-runtime/blob/4696b4822fc83d914bf54d4422e2c93f6adf505f/runtime/environment-cpu.yml#L11

jayqi commented on runtime/environment-cpu.yml:
```diff
@@ -38,3 +38,8 @@ dependencies:
+      - imblearn==0.0
+      - gensim==3.8.3
+      - tensorboard-logger==0.1.0
+      - futures==2.1.6
```
[futures](https://pypi.org/project/futures/) is a backport of `concurrent.futures` to Python 2. We are running Python 3.9, so this doesn't seem like it would make sense. Can you please clarify why we need this?

jayqi commented on runtime/environment-cpu.yml:
```diff
@@ -38,3 +38,8 @@ dependencies:
       - tensorflow-probability==0.15.0
+      - deepwalk==1.0.3
+      - imblearn==0.0
+      - gensim==3.8.3
```
This version of gensim has been included as part of another PR that has been merged in, so you should be all set on this one. See #33.

jayqi commented on runtime/environment-cpu.yml:
```diff
@@ -38,3 +38,8 @@ dependencies:
+      - deepwalk==1.0.3
+      - imblearn==0.0
+      - gensim==3.8.3
+      - tensorboard-logger==0.1.0
```
When your submission runs on the hosted infrastructure, you won't be able to get any of the tensorboard events out from that run, so I don't think this will be very useful.

Would it be the case that you could use this for only local experimentation in a local image where you've added this? Or is use of it heavily integrated throughout your code?

jayqi commented:
Hi @ambrishrawat, I've left a few comments. A few of your requests are already in, and a few others I'd like some clarifications from you.

Regarding igraph, per @szhorvat's note, it would be worth trying to understand what errors you are seeing and whether they are easily addressable. (Thanks for the input, @szhorvat!)

@mahmoudsajjadi was the other participant who had originally requested igraph 0.10.3 (and verified that igraph 0.10.2 worked fine). Do you have any experience trying 0.9.11, @mahmoudsajjadi?

mahmoudsajjadi commented:
> Hi @ambrishrawat, I've left a few comments. A few of your requests are already in, and a few others I'd like some clarifications from you.
> 
> Regarding igraph, per @szhorvat's note, it would be worth trying to understand what errors you are seeing and whether they are easily addressable. (Thanks for the input, @szhorvat!)
> 
> @mahmoudsajjadi was the other participant who had originally requested igraph 0.10.3 (and verified that igraph 0.10.2 worked fine). Do you have any experience trying 0.9.11, @mahmoudsajjadi?

Hi @jayqi, Our code works well with igraph 0.10.3 and 0.10.2 but it seems there was some problem with version 0.10.1.  We only experiment with updated versions.

ambrishrawat commented:
Hi @jayqi , I have incorporated your suggestions in the comments and updated the branch. Thank you @szhorvat for the link, we were able to resolve the issue with the `igraph` version.

szhorvat commented:
Just wanted to note that python-igraph 0.10.3 is now available on conda-forge, in case you want to update fro 0.10.2.
